### DISCUSSION TITLE
Webhook: access "snapshots" data in conditions

### DISCUSSION THREAD

**Discussion by cs-1** [netbox-community/netbox#11390] [Q&A]
Hi all!

In webhooks you can use `snapshots` data in Jinja2 templates. My question is: is there any way to access `snapshots` attributes in the Conditions JSON of a webhook? Use case: I'd like to write a condition that is true when a device is renamed. It could look something like this:

```json
{
 "attr": "name",
 "value": "{{ snapshots['postchange']['name'] }}",
 "negate": true
}
```

According to the docs this isn't possible. Is there any other way to catch a renaming event in a condition?

**Comment by bkampsnl** [comment]
Is this possible already ? I also would like to use prechange / postchange data in webhook conditions..

**Comment by bitcollector1** [comment]
I'm also wondering if this is possible?

**Comment by VladyslavTkachenkoDesy** [comment]
hey, any updates here?

**Comment by Hardi100** [comment]
I think only the data gets validated, not the snapshots. 
I think this is the line, that validates the condition: https://github.com/netbox-community/netbox/blob/develop/netbox/extras/events.py#L90
It would be a nice feature and could help to reduce the actions that are triggered unnecessarily.
Can someone create a feature request?

**Reply by Hardi100** [reply]
https://github.com/netbox-community/netbox/issues/14896 this could be similar
